tests/test_fqdn.py

```diff
@@ -1,183 +1,219 @@
+    def test_relative_fqdn_true(self, strict):
+        assert build_fqdn("relative.com", strict).is_valid_relative is True
 
-    def test_relative_fqdn_false(self):
-        assert FQDN("relative.com.").is_valid_relative is False
+    def test_relative_fqdn_false(self, strict):
+        assert build_fqdn("relative.com.", strict).is_valid_relative is False
 
 
-class TestRelativeFQDN(TestCase):
-    def test_relative_fqdn_from_relative(self):
-        self.assertEqual(FQDN("trainwreck.com").relative, "trainwreck.com")
+class TestRelativeFQDN:
+    def test_relative_fqdn_from_relative(self, strict):
+        assert build_fqdn("trainwreck.com", strict).relative == "trainwreck.com"
 
-    def test_relative_fqdn_from_absolute(self):
-        self.assertEqual(FQDN("trainwreck.com.").relative, "trainwreck.com")
+    def test_relative_fqdn_from_absolute(self, strict):
+        assert build_fqdn("trainwreck.com.", strict).relative == "trainwreck.com"
 
-    def test_relative_fqdn_from_invalid(self):
-        with self.assertRaises(ValueError):
-            FQDN("trainwreck..").relative
+    def test_relative_fqdn_from_invalid(self, strict):
+        with pytest.raises(ValueError):
+            build_fqdn("trainwreck..", strict).relative
 
 
-class TestEquality(TestCase):
-    def test_absolutes_are_equal(self):
-        self.assertEqual(FQDN("trainwreck.com."), FQDN("trainwreck.com."))
+class TestEquality:
+    def test_absolutes_are_equal(self, strict):
+        assert build_fqdn("trainwreck.com.", strict) == build_fqdn(
+            "trainwreck.com.", strict
+        )
 
-    def test_relatives_are_equal(self):
-        self.assertEqual(FQDN("trainwreck.com"), FQDN("trainwreck.com"))
+    def test_relatives_are_equal(self, strict):
+        assert build_fqdn("trainwreck.com", strict) == build_fqdn(
+            "trainwreck.com", strict
+        )
 
-    def test_mismatch_are_equal(self):
-        self.assertEqual(FQDN("trainwreck.com."), FQDN("trainwreck.com"))
+    def test_mismatch_are_equal(self, strict):
+        assert build_fqdn("trainwreck.com.", strict) == build_fqdn(
+            "trainwreck.com", strict
+        )
 
-    def test_equality_is_case_insensitive(self):
-        self.assertEqual(
-            FQDN("all-letters-were-created-equal.com."),
-            FQDN("ALL-LETTERS-WERE-CREATED-EQUAL.COM."),
+    def test_equality_is_case_insensitive(self, strict):
+        assert build_fqdn("all-letters-were-created-equal.com.", strict) == build_fqdn(
+            "ALL-LETTERS-WERE-CREATED-EQUAL.COM.", strict
         )
 
+    def test_strict_and_loose_can_be_equal(self):
+        assert FQDN("trainwreck.com.", True) == FQDN("trainwreck.com", False)
+
 
-class TestHash(TestCase):
-    def test_is_hashable(self):
-        self.assertTrue(hash(FQDN("trainwreck.com.")))
+class TestHash:
+    def test_is_hashable(self, strict):
+        assert hash(build_fqdn("trainwreck.com.")), strict is True
```

Comment:
missing equality check

This passes because assert works like this:
`assert condition, "error message"`